x-pack/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {
+  Position,
+  Chart,
+  Settings,
+  Axis,
+  LineSeries,
+  getAxisId,
+  getSpecId,
+  AreaSeries,
+  BarSeries,
+} from '@elastic/charts';
+import { Datatable } from '../../../canvas/canvas_plugin_src/functions/types';
+
+/**
+ * This file contains TypeScript type definitions and their equivalent expression
+ * definitions, for configuring and rendering an XY chart. The XY chart serves
+ * triple duty as a bar, line, or area chart.
+ *
+ * The xy_chart expression function serves mostly as a passthrough to the xy_chart_renderer
+ * which does the heavy-lifting.
+ */
+
+/**
+ * Configuration for the chart's legend.
+ */
+export interface LegendConfig {
+  isVisible: boolean;
+  position: Position;
+}
+
+/**
+ * LegendConfig as an expression type.
+ */
+export const legendConfig = {
+  name: 'legendConfig',
+  aliases: [],
+  type: 'legendConfig',
+  help: `Configure a chart's legend`,
+  context: {
+    types: ['null'],
+  },
+  args: {
+    isVisible: { types: ['boolean'] },
+    position: { types: ['string'] },
+  },
+  fn: function fn(_context: any, args: LegendConfig) {
+    return {
+      type: 'legendConfig',
+      ...args,
+    };
+  },
+};
```

Comment:
I was expecting a single function to be registered to handle all xy_chart options, and just to confirm my understanding what this will look like when combined is something like this:

```
| xy_chart
  legend={legendConfig isVisible=true position=left}
  x={xConfig accessors=[col-1]}
  y={yConfig accessors=[col-0]}
  stackAccessors=[col-0]
```

I see that there is something similar in canvas today called `axisConfig` which is a function that configures a single axis. Is there any overlap?

Also, I noticed that you're not using any of the existing typescript support for canvas. If you look at AxisConfig for example, it's defined as `ContextFunction<'axisConfig', Datatable, Arguments, AxisConfig>`

Reply:
We don't want to pull in any types from `x-pack/plugins/canvas...` just because those are really internal Canvas concerns. In fact, I need to change this line `import { Datatable } from '../../../canvas/canvas_plugin_src/functions/types';`. I wonder if we should move the Datatable type definition into the core interpreter.

Anyway, if we decide that it's worth it, we could pull shared types / functions / modules out of Canvas into some common location such as the core interpreter. In our case, we want to be able to fairly easily move some things from x-pack to OSS, and I think (at least for now), it makes the most sense for us to own our own definitions and types and minimize dependencies.